pgb_utils/pgb_utils/utils.py

```diff
@@ -35,7 +36,19 @@ def alert_dict_to_dataframe(alert_dict: dict) -> pd.DataFrame:
 
 def alert_dict_to_table(alert_dict: dict) -> Table:
     """Package a ZTF alert dictionary into an Astopy Table."""
-    rows = [alert_dict["candidate"]] + alert_dict["prv_candidates"]
+    candidate = OrderedDict(alert_dict["candidate"])  # dict
+    cand_cols = set(candidate.keys())
+
+    # astropy 3.2.1 cannot handle dicts with different keys (fixed by v4.1)
+    prv_candidates = []  # recreate with missing keys added
+    for prv_cand in alert_dict["prv_candidates"]:
+        prvc_cols = set(prv_cand.keys())
+        missing_cols = cand_cols - prvc_cols
+        empty = {mc: None for mc in missing_cols}
+        tmp = {**prv_cand, **empty}
+        prv_candidates.append(tmp)
+
+    rows = [candidate] + prv_candidates
```

Comment:
This is O(n). Using `.append` would be O(1).

Reply:
I haven't read the context here closely ,but I believe `.extend` would be the equivalent operation rather than `.append`.

Reply:
Ah, I hadn't appreciated that the `candidate` element needed to be the first in the list.

Reply:
I was thinking of `prv_candidates.append(candidate)`

Reply:
@djperrefort are you advocating for creating a temporary array and then extending that?
```
candidate_tmp = [candidate]
rows = candidate_tmp.extend(prv_candidate)
```

Reply:
I'm advocating that the list in the above accumulation pattern with `candidate` already in it. I.e., `prv_candidates = [candidate]` before the for loop.

Reply:
I agree - optimizing this call won't yield a measurable performance gain. I'm mostly pushing on this from a readability point of view. I probably should have merged this comment with my comment on the above for loop.

Reply:
I updated the code to instantiate the list with `candidate` already in it.
Thanks for the pointers.